hudi-common/src/main/java/org/apache/hudi/storage/HoodieStorageUtils.java

```diff
@@ -44,4 +44,27 @@ public static HoodieStorage getStorage(StoragePath path, StorageConfiguration<?>
       throw new HoodieException("Unable to create " + storageClass, e);
     }
   }
+
+  /**
+   * Creates a {@link HoodieStorage} instance using custom constructor parameters.
+   * This method uses reflection to instantiate the storage class with provided parameter types.
+   *
+   * @param conf       storage configuration containing the storage class name
+   * @param paramTypes array of parameter types for the constructor
+   * @param params     array of parameter values for the constructor
+   * @return {@link HoodieStorage} instance
+   * @throws HoodieException if unable to create the storage instance
+   */
+  public static HoodieStorage getStorage(StorageConfiguration<?> conf,
```

Comment:
I don't think we should add this method. This will fail all storage implementations that don't have a constructor that takes `params`. 

Looking at other changes, it seems like only purpose of this method is to instantiate `HoodieHadoopStorage` when you only have `FileSystem` available and don't even have a path. These cases can stay using `HoodieHadoopStorage` constructor imo